miniapp/miniapp_gen_eigensolver.cpp

```diff
@@ -94,7 +122,18 @@ struct GenEigensolverMiniapp {
     GlobalElementSize matrix_size(opts.m, opts.m);
     TileElementSize block_size(opts.mb, opts.mb);
 
-    ConstHostMatrixType matrix_a_ref = [matrix_size, block_size, &comm_grid]() {
+    // Default capture & is needed to suppress warning of unused opts when DLAF_WITH_HDF5 is not defined.
+    // [[maybe_unused]] is not supported in lambda captures
```

Comment:
No big preference either way, but I slightly prefer an explicit version with an `#else` branch using `silenceUnusedWarningFor`. This would avoid the use ofa catch-all `&`. Not a blocking comment.

Reply:
I forgot about this! Thanks for pointing it out. Let's change it.